packages/okta-react/src/SecureRoute.js

```diff
@@ -10,19 +10,20 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useOktaAuth } from './OktaContext';
-import { useHistory, Route } from 'react-router-dom';
+import { Route } from 'react-router-dom';
 
 const RequireAuth = ({ children }) => { 
   const { authService, authState } = useOktaAuth();
-  const history = useHistory();
 
-  if(!authState.isAuthenticated) { 
-    if(!authState.isPending) { 
-      const fromUri = history.createHref(history.location);
-      authService.login(fromUri);
-    }
+  useEffect(() => {
+    if(!authState.isAuthenticated && !authState.isPending) { 
+      authService.login();
```

Comment:
Is this where it is possible to begin concurrent login attempts? If the authState is updated (due to token changes, etc). Do we need to add aa concurrent check on this or place this call into the PromiseQ?

Reply:
Yes. Probably the `isPending` state is not enough here as it's triggered by emitting an event, which may still run into concurrent flow.

Reply:
Resolved by providing pending state in the login method to make sure it only be called once.